chapters/4/architektura.tex

```diff
@@ -0,0 +1,43 @@
+% LTex: language=pl
+\section{Architektura wdrożenia}
+\subsection{Topologia logiczna}
+Wraz ze zmianą technologii, na której oparta była architektura wdrożenia systemu STOS, zmianie musiała ulec również architektura rozwiązania.
+
+\begin{figure}[!h]
+	\begin{center}
+		\resizebox{0.7\textwidth}{!} {
+			\includegraphics{img/4/wdrozenie.png}
+		}
+		\caption[Topologia wdrożenia systemu STOS]{Diagram przedstawiający topologię wdrożenia platformy serwerowej systemu STOS opartego na wirtualizacji oferowanej przez xcp-ng. Źródło własne.}
+		\label{diagramTopologiaStos}
+	\end{center}
+\end{figure}
+
+\noindent Diagram \ref{diagramTopologiaStos} przedstawia klaster i komponenty wdrożonego systemu STOS jako maszyny wirtualne. W ramach rozpatrywania komponentów samego systemu należy zwrócić uwagę na trzy główne rodzaje maszyn: \textit{STOS}, \textit{STOS-dev} i \textit{STOS-logs}. W porównaniu z proponowanym rozwiązaniem opartym o zastosowanie klastra Kubernetes można zaobserwować diametralnie zmniejszoną liczbę osobnych bytów z perspektywy architektury wdrożenia. Z racji o wiele większych kosztów związanych z wdrożeniem i uruchomieniem maszyny wirtualnej w porównaniu z kontenerem, logiczne grupowanie serwisów i ich enkapsulacja odbywa się na poziomie całej maszyny wirtualnej, co pozwala na logiczne rozdzielenie części systemu odpowiedzialnych za podobne funkcjonalności, jak i uproszczenie protokołu komunikacji między serwisami, które ściśle ze sobą współpracują. Na szczególną uwagę zasługuje zmiana w architekturze głównych komponentów platformy serwerowej systemu STOS -- serwisu zarządzającego \textit{"manager"} i modułu kompilującego \textit{"worker"}. Dzięki wdrożeniu ich na jednej maszynie wirtualnej zniknęła konieczność korzystania z protokołu http w ramach komunikacji pomiędzy tymi dwoma serwisami, dzięki czemu złożoność programów uległa zmniejszeniu. Podobna zmiana dotyczy funkcjonalności agregacji logów z systemu STOS, ponieważ wszystkie serwisy odpowiedzialne za procesowanie logów zostały umieszczone w ramach jednej maszyny wirtualnej.
```

Comment:
W manager trzeba zastosować cudzysłów otwierający „manager” - może to da radę w latexie ustawić